lib/internal/errors.js

```diff
@@ -821,8 +821,14 @@ function hideInternalStackFrames(error) {
 // to make usage of the error in userland and readable-stream easier.
 // It is a regular error with `.code` and `.name`.
 class AbortError extends Error {
-  constructor() {
-    super('The operation was aborted');
+  constructor(message = 'The operation was aborted', options = {}) {
+    if (options === null || typeof options !== 'object') {
+      throw new codes.ERR_INVALID_ARG_TYPE('options', 'Object', options);
+    }
+    const {
+      cause,
+    } = options;
+    super(message, { cause });
```

Comment:
The spec doesn't require `options` to be an object, maybe we shouldn't either.
```suggestion
  constructor(message = 'The operation was aborted', options = undefined) {
    super(message, { cause: options?.cause });
```
or
```suggestion
  constructor(message = 'The operation was aborted', options = undefined) {
    super(message, options);
```

Reply:
I'd rather have the stricter type checking on the `options` object itself given that it's a node.js defined object.

Reply:
Alright, we could still avoid creating a throw away object
```suggestion
  constructor(message = 'The operation was aborted', options = undefined) {
    if (options !== undefined && typeof options !== 'object') {
      throw new codes.ERR_INVALID_ARG_TYPE('options', 'Object', options);
    }
    super(message, { cause: options?.cause });
```
or even
```suggestion
  constructor(message = 'The operation was aborted', options = undefined) {
    if (options !== undefined && typeof options !== 'object') {
      throw new codes.ERR_INVALID_ARG_TYPE('options', 'Object', options);
    }
    super(message, options);
```